python_modules/dagster/dagster_tests/core_tests/structured_config_tests/test_basic_structured_config.py

```diff
@@ -136,6 +137,28 @@ def my_asset(config: AnAssetConfig):
     )
 
 
+def test_multi_asset():
+    class AMultiAssetConfig(Config):
+        a_string: str
+        an_int: int
+
+    @multi_asset(outs={"a": AssetOut(key="asset_a"), "b": AssetOut(key="asset_b")})
+    def two_assets(config: AMultiAssetConfig):
+        assert config.a_string == "foo"
+        assert config.an_int == 2
+        return 1, 2
+
+    assert (
+        build_assets_job(
+            "blah",
+            [two_assets],
+            config={"ops": {"two_assets": {"config": {"a_string": "foo", "an_int": 2}}}},
+        )
+        .execute_in_process()
+        .success
+    )
+
+
```

Comment:
One thing I like to do in test cases like these is to absolutely ensure that the function body has been executed. 

Meaning before the function (in this case the asset):

`executed = {}`

Then in the function body:

`executed["yes"] = True`

Then after job execution:

`assert executed["yes"]`

When the next engineer comes along and they see that they have high confidence that the code is actually executed. Whenever I work on a framework that is calling back into user code that is doing asserts (and not verifying outputs) I usually do this. More than once I've been bitten debugging something only realizing after some time that some function was actually never called.

Don't feel super strongly so I'm not going to block commit, but I generally find it's a good best practice.

Reply:
updated